unitTests/controllers/QueryControllerTest.ts

```diff
@@ -461,5 +463,64 @@ export function QueryControllerTest() {
         });
       });
     });
+
+    describe('coveo-lastQueryHash local storage', () => {
+      const key = 'coveo-lastQueryHash';
+      const fakeResults: () => IQueryResults = () => ({ ...FakeResults.createFakeResults(), pipeline: 'commerce', searchUid: '123' });
+      const expireTime = () => new Date().getTime() + 1000 * 60 * 30;
+      beforeEach(() => {
+        localStorage.clear();
+        (test.env.searchEndpoint.search as jasmine.Spy).and.returnValue(Promise.resolve(fakeResults()));
+      });
+
+      it(`with no local storage value
+      after a query has been executed
+      calling saveLastQuery should save the last query hash to the local storage`, async done => {
+        await test.cmp.executeQuery();
+        test.cmp.saveLastQuery();
+
+        const queryHash = JSON.parse(localStorage.getItem(key));
+        expect(queryHash).toEqual(
+          jasmine.objectContaining({
+            expire: jasmine.any(Number),
+            uid: '123'
+          })
+        );
+        expect(JSON.parse(queryHash.hash)).toEqual({ pipeline: 'commerce', sortCriteria: 'relevancy' });
+        done();
+      });
+
+      it(`with a local storage value
+      after a query has been executed using the same parameters
+      the local storage item should be loaded 
+      and "original" search uid should be reused`, async done => {
+        const hash = JSON.stringify({ pipeline: 'commerce', sortCriteria: 'relevancy' });
+        const value = JSON.stringify({ expire: expireTime(), uid: '456', hash });
+        localStorage.setItem(key, value);
+
+        const results = await test.cmp.executeQuery();
+
+        expect(results.searchUid).toBe('456');
+        expect(results._reusedSearchUid).toBe(true);
+        expect(localStorage.getItem(key)).toBeFalsy();
+        done();
+      });
+
+      it(`with a local storage value
+      after a query has been executed using different parameters
+      the local storage item should be loaded 
+      but the "new" search uid should be used`, async done => {
```

Comment:
Hmm, this phrase in test description is not clear: `the local storage item should be loaded`
What does `loaded` mean here? Is it updating the values to those in the most recent query?

It also says `the "new" search uid should be used`.
Wouldn't this be `456` on line 514? vs. the assertion on line 519 on `123`?

Reply:
loaded has read and deleted (method is also called loadLastQueryHash, which calls this.localStorage.load()) I think load is appropriate

The new one is 123, which the api returns (test.cmp.executeQuery()) and not the saved Id